epubmaker_preprocessing.rb

```diff
@@ -343,18 +341,32 @@ def addHtmlLineBreaks(filecontents, logkey='')
 @log_hash['linkauthorarr'] = linkauthorarr
 @log_hash['linkauthorid'] = linkauthorid
 
+# uncomment newsletter link placeholders
+filecontents = activateAuthorLinks(myhash, filecontents, 'uncomment_newsletter_links')
+
+# write epub-ready html to file
+overwriteFile(epub_tmp_html, filecontents, 'overwrite_epubhtml_final')
+
 # update newsletter author links, for single or multiple authors
-filecontents = setNewsletterAuthorLinks(linkauthorarr, linkauthorid, filecontents, myhash, 'set_newsletter_auth_links')
+if linkauthorarr.count > 1
+  filecontents = setNewsletterAuthorLinksMultiple(linkauthorarr, linkauthorid, myhash, newsletterjs, epub_tmp_html, 'set_newsletter_auth_links_multiple')
+else
+  filecontents = setNewsletterAuthorLinksSingle(linkauthorarr, linkauthorid, myhash, newslettersinglejs, epub_tmp_html, 'set_newsletter_auth_links_single')
+end
 
 #replace imprint, eisbn placeholders
-filecontents = updateImprintandEisbnPlaceholders(myhash, filecontents, 'update_eisbn_&_imprint_placeholders')
+filecontents = updateImprintandEISBNPlaceholders(myhash, filecontents, 'update_eisbn_&_imprint_placeholders')
 
 # add some line breaks to make the html easier to deal with
 filecontents = addHtmlLineBreaks(filecontents, 'add_html_line_breaks')
 
 # write epub-ready html to file
 overwriteFile(epub_tmp_html, filecontents, 'overwrite_epubhtml_final')
 
+# FOR TESTING ONLY; DELETE WHEN DONE
```

Comment:
You were probably going to delete lines 366-368 already anyways, but noting just in case ;)